### ISSUE TITLE
Phase 3: Create and Configure GEMMA Tuning Workflow (Otomasyon Pipeline'ı Kurulumu)

### ISSUE THREAD

**Issue by SefaGH** [SefaGH/bearish-alpha-bot#375]
### 🎯 Amaç (Goal)

Bu görevin temel amacı, FAZ 1'de doğrulanan ve FAZ 2'de GEMMA için uyarlanan tüm veri hazırlama, özellik seçimi ve hiperparametre optimizasyonu adımlarını otomatize eden yeni bir GitHub Actions workflow'u oluşturmaktır.

Bu amaçla, başarılı `full-lstm-tuning.yml` workflow'u bir şablon olarak kullanılacak ve **`.github/workflows/full-gemma-tuning.yml`** adında yeni bir workflow dosyası yaratılacaktır. Bu workflow, tüm "Ar-Ge" sürecini baştan sona yürütecek ve sonuçları bir "artifact" olarak sunacaktır.

### ⚙️ Referans ve Strateji (Reference & Strategy)

*   **Ana Şablon (Blueprint):** Bu görev için tek ve yegane referans dosyası **`.github/workflows/full-lstm-tuning.yml`**'dir. Yeni bir yapı kurma; bu kanıtlanmış otomasyon şablonunu kopyala ve GEMMA için gerekli değişiklikleri yap.

#### **Adım 1: Yeni Workflow Dosyasını Oluşturma**

1.  Mevcut `.github/workflows/full-lstm-tuning.yml` dosyasının içeriğinin tamamını kopyala.
2.  Bu içeriği, **`.github/workflows/full-gemma-tuning.yml`** adında yeni bir dosyaya yapıştır. Aşağıdaki tüm değişiklikler bu yeni dosya üzerinde yapılacaktır.

#### **Adım 2: Workflow Ayarlarını ve İsimlendirmeleri Güncelleme**

Yeni `.github/workflows/full-gemma-tuning.yml` dosyası içinde aşağıdaki genel ayarları güncelle:

1.  Workflow'un en üstündeki `name` alanını, ayırt edici olacak şekilde değiştir.
    *   **Değiştir:** `name: 🚀 Full LSTM Tuning (Balanced Accuracy)`
    *   **Yeni Değer:** `name: 💎 GEMMA - Full Hyperparameter Tuning`

2.  `jobs` bölümü altındaki `full-tuning` işinin `name` alanını güncelle.
    *   **Değiştir:** `name: 🎯 Full LSTM Hyperparameter Tuning (Balanced Accuracy)`
    *   **Yeni Değer:** `name: 🎯 Full GEMMA Hyperparameter Tuning (MLP)`

#### **Adım 3: Optimizasyon Adımını (Tuning Step) GEMMA İçin Değiştirme**

Bu, görevin en kritik adımıdır. LSTM optimizasyonunu çalıştıran adımı, FAZ 2'de oluşturduğumuz GEMMA optimizasyon script'ini çalıştıracak şekilde değiştireceğiz.

1.  Workflow dosyasında `🚀 Step 4 - Full LSTM Tuning` adıyla veya benzer bir isimle bilinen adımı (`step`) bul.
2.  Bu adımın `name` ve `run` alanlarını aşağıdaki gibi güncelle:

    ```yaml
    # Bu adımı bul ve içeriğini değiştir
    - name: 🚀 Step 4 - Full GEMMA Tuning
      run: |
        echo "======================================================================"
        echo "🚀 FULL GEMMA HYPERPARAMETER TUNING (MLP MODEL)"
        echo "======================================================================"
        echo ""
        echo "Configuration:"
        echo "  - Model: GEMMA (MLP)" # <-- DEĞİŞTİ
        echo "  - Trials: ${{ github.event.inputs.trials }}"
        echo "  - CV Splits: ${{ github.event.inputs.cv_splits }}"
        echo "  - Symbol: ${{ github.event.inputs.symbol }}"
        echo "  - Optimization Metric: balanced_accuracy_score"
        echo ""
        echo "This will take approximately 45-60 minutes..."
        echo ""
        
        # ==================== DEĞİŞEN ANA KOMUT BURASI ====================
        python scripts/tune_gemma_model_standalone.py \
          --model gemma \
          --symbol '${{ github.event.inputs.symbol }}' \
          --trials ${{ github.event.inputs.trials }} \
          --cv-splits ${{ github.event.inputs.cv_splits }}
        # ================================================================
        
        echo ""
        echo "✅ Full GEMMA tuning completed!"
    ```
    *   **Önemli Değişiklikler:**
        *   Adımın adı `Full GEMMA Tuning` olarak değiştirildi.
        *   `echo` komutları, sürecin GEMMA için olduğunu yansıtacak şekilde güncellendi.
        *   En önemlisi, çalıştırılan Python script'i `tune_regime_models_standalone.py`'den **`tune_gemma_model_standalone.py`**'ye değiştirildi.
        *   Script'e verilen `--model` argümanı `lstm`'den **`gemma`**'ya değiştirildi.

#### **Adım 4: Artifact Yükleme Adımını Güncelleme**

Workflow'un sonunda sonuçları artifact olarak yükleyen adımı, GEMMA'ya özel ve ayırt edici olacak şekilde güncelle.

1.  `💾 Upload Results` adlı adımı bul.
2.  Bu adımın `with` bloğu altındaki `name` alanını güncelle.

    ```yaml
    # Bu adımı bul ve 'name' alanını değiştir
    - name: 💾 Upload Results
      uses: actions/upload-artifact@v4
      if: always()
      with:
        name: gemma-tuning-results-${{ github.run_number }} # <-- DEĞİŞTİ
        path: |
          logs/tuning_results/gemma_tuning_*.json # <-- DEĞİŞTİ
          logs/*.log
        retention-days: 30
    ```
    *   **Önemli Değişiklikler:**
        *   Artifact adı, `tuning-results-` yerine **`gemma-tuning-results-`** oldu.
        *   Yüklenecek `.json` dosyasının yolu, `lstm_tuning_*.json` yerine **`gemma_tuning_*.json`** oldu. Bu, FAZ 2'de yaptığımız isimlendirme değişikliğiyle uyumludur.

### ✅ Başarı Kriterleri (Code-Level Success Criteria)

- [ ] `.github/workflows/full-gemma-tuning.yml` adında yeni bir workflow dosyası oluşturulmuş olmalıdır.
- [ ] Yeni workflow'un `name` ve `job name` alanları "GEMMA" kelimesini içerecek şekilde güncellenmiş olmalıdır.
- [ ] "Tuning" adımında çalıştırılan script'in adı **`scripts/tune_gemma_model_standalone.py`** olarak doğru bir şekilde değiştirilmiş olmalıdır.
- [ ] Bu script'e verilen `--model` argümanı **`gemma`** olmalıdır.
- [ ] Sonuçları yükleyen `upload-artifact` adımının `name` ve `path` alanları, **`gemma`** ön ekini kullanacak şekilde doğru bir şekilde güncellenmiş olmalıdır.
- [ ] Workflow dosyasındaki diğer adımlar (veri hazırlama, özellik analizi, scaler kaydetme) **değiştirilmeden** korunmalıdır, çünkü bu adımlar modelden bağımsızdır ve GEMMA için de gereklidir.
- [ ] Kod, geçerli bir YAML formatında olmalı ve herhangi bir çalıştırma gerektirmemelidir.

**Comment by SefaGH** [comment]
#### **Critical Action: Environment Setup and Verification**

1.  **Enforce Python Version 3.11:**
    *   The project requires Python `3.11.x`. The runner environment may default to a different version (e.g., 3.12).
    *   **Action:** Your first step must be to check the active Python version (`python --version`). If it is not `3.11`, you must stop, install Python 3.11, and switch the environment to use it exclusively.

2.  **Mandatory Full Dependency Installation:**
    *   The validation requires all project dependencies, including heavy libraries like `torch`, `pandas`, and `scikit-learn`. Skipping any package will lead to invalid test results.
    *   **Action:** After ensuring Python 3.11 is active, perform a complete and non-simplified installation of all packages listed in the project's dependency files (e.g., `requirements.txt`). Do not use mocked libraries for this validation.

---

#### **Execution Directive**

Once the environment is confirmed to be running **Python 3.11** and **all dependencies are fully installed**, proceed with executing the complete validation plan as detailed in the issue description above, starting from `Task 1`.

The goal is a full, non-simplified validation of the artifacts and code.
